mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/MicFragment.kt

```diff
@@ -1,17 +1,80 @@
 package rs.readahead.washington.mobile.views.fragment
 
+import android.Manifest
+import android.animation.AnimatorInflater
+import android.animation.ObjectAnimator
+import android.content.Context
+import android.content.pm.PackageManager
 import android.os.Bundle
-import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.ImageButton
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.core.app.ActivityCompat
+import com.hzontal.tella_vault.VaultFile
+import io.reactivex.disposables.CompositeDisposable
+import org.hzontal.shared_ui.bottomsheet.BottomSheetUtils
 import rs.readahead.washington.mobile.R
+import rs.readahead.washington.mobile.media.AudioRecorder
+import rs.readahead.washington.mobile.media.AudioRecorder.AudioRecordInterface
+import rs.readahead.washington.mobile.media.MediaFileHandler
+import rs.readahead.washington.mobile.mvp.contract.IAudioCapturePresenterContract
+import rs.readahead.washington.mobile.mvp.contract.IMetadataAttachPresenterContract
+import rs.readahead.washington.mobile.mvp.contract.ITellaFileUploadSchedulePresenterContract
+import rs.readahead.washington.mobile.mvp.presenter.AudioCapturePresenter
+import rs.readahead.washington.mobile.mvp.presenter.MetadataAttacher
+import rs.readahead.washington.mobile.mvp.presenter.TellaFileUploadSchedulePresenter
+import rs.readahead.washington.mobile.util.DateUtil.getDateTimeString
+import rs.readahead.washington.mobile.util.StringUtils
+import rs.readahead.washington.mobile.views.base_ui.MetadataBaseLockFragment
+import timber.log.Timber
+
+import java.util.*
+import java.util.concurrent.TimeUnit
 
 private const val ARG_PARAM1 = "param1"
 private const val ARG_PARAM2 = "param2"
 
-class MicFragment : Fragment() {
+const val RECORD_REQUEST_CODE = 2001
+
+class MicFragment : MetadataBaseLockFragment(), AudioRecordInterface,
+    IAudioCapturePresenterContract.IView,
+    ITellaFileUploadSchedulePresenterContract.IView,
+    IMetadataAttachPresenterContract.IView {
+
     // TODO: Rename and change types of parameters
+    private val TIME_FORMAT: String = "%02d:%02d"
+    //var RECORDER_MODE = "rm"
+
+    private var animator: ObjectAnimator? = null
+
+    private var notRecording = false
+
+    private val UPDATE_SPACE_TIME_MS: Long = 60000
+    private var lastUpdateTime: Long = 0
+
+    // handling MediaFile
+    private var handlingMediaFile: VaultFile? = null
+
+    // recording
+    private var uploadPresenter: TellaFileUploadSchedulePresenter? = null
+    private val disposable = CompositeDisposable()
+    var presenter: AudioCapturePresenter? = null
```

Comment:
`uploadPresenter`, `presenter`, `audioRecorder` could be initialized as **lazy** like this 

`private val audioRecorder  by lazy { AudioCapturePresenter(this)  }` like this you are sure that the object will be created only when it is called, and you will always use the same object in the current scope.